index.js

```diff
@@ -61,28 +64,20 @@ export default store => {
 
     } else {
 
-      //finish displatching the action
+      //finish dispatching the action
       const result = next(action);
 
       //get the updated state
       const state = store.getState();
 
-      waiting.forEach((when, index) => {
-
-        //check if the condition is met
-        if (when.payload.condition(state, action)) {
-
-          //remove the delayed action
-          if (when.type === ONCE) {
-            waiting.splice(index, 1);
-          }
-
-          //dispatch the delayed action
-          store.dispatch(when.payload.createAction(action));
+      const readyToBeDispatched = waiting
```

Comment:
Ah! I changed it back because I thought the `for` loop was more readable and saved a few cpu cycles. If it is an issue with JS arrays please feel free to change it back!